Mock tests

MaxGfeller opened the pull request:
This PR enables users to test the library without having to have a Killbill instance running at `127.0.0.1:8080`. It should also enable tests on Travis-CI.

The mocks themselves can be recorded with `composer record-mocks`. This will first run an integration test and obviously needs a running Killbill instance. Make sure to start with an empty Killbill database, otherwise it will fail.

To do an integration test (exact same test but against a running instance) just run `composer test-integration`.

MaxGfeller commented:
Seems like older and newer versions of PHP still do have some problems. I'm working on it.

MaxGfeller commented:
Alright, PHP versions `5.3`,  `5.4` and `5.5` are officially [End Of Life](https://secure.php.net/supported-versions.php). I think we could throw those out in `.travis.yml`?

Still looking into the errors with PHP >7.

pierre commented on test/KillbillTest.php:
```diff
@@ -32,8 +32,14 @@ class KillbillTest extends \PHPUnit_Framework_TestCase
     {
+        Client::$mockManager = new MockManager();
+        $externalKey = uniqid();
+        if (getenv('ENV') === 'local' || getenv('RECORD_REQUESTS') == '1') {
```
Could this logic be extracted in a utility function, since it's needed in various places?

MaxGfeller replied:
I rewrote that part a bit, see newest changes.

pierre commented on src/Client.php:
```diff
@@ -51,16 +52,24 @@ public function request($method, $uri, $data = null, $user = null, $reason = nul
 
     private function _sendRequest($method, $uri, $data = null, $user = null, $reason = null, $comment = null, $additional_headers = null) {
 
+        if (getenv('ENV') === 'local' && isset(self::$mockManager)) {
```
Would this impact overall performance, since it would be called in a production environment for each HTTP call?

Also, and I'm unsure if there is a better way to do it, it's a bit unfortunate to have test specific code in the main codepath.

pierre commented on README.md:
```diff
@@ -66,6 +66,12 @@ If you want to use the client but are not using [Composer](https://getcomposer.o
+Testing
+-------
+
+The Killbill PHP client uses [phpunit](https://phpunit.de/) as testing framework. Tests can either be run locally using `npm test` or against
```
Is there any setup required for npm to work (e.g. `package.json`)?

MaxGfeller replied:
Whoops, i'm sorry. This should've actually been `composer test`. No need to install `npm` or `node`.

Too much context switching 🙈

MaxGfeller commented:
Yay, tests now work in PHP 7 too 🎉  The problem was that the object keys did not get sorted correctly.

pierre commented:
:1st_place_medal:

MaxGfeller commented:
Awesome, thanks for merging! Could you tag a new version for packagist? Preferrably `0.1.0` because there were a lot of breaking changes since the last release.

pierre commented:
Done! https://github.com/killbill/killbill-client-php/releases/tag/0.1.0
